code/datums/craft/gun_parts.dm

```diff
@@ -0,0 +1,262 @@
+/obj/item/part/gun/frame
+	name = "gun frame"
+	desc = "a generic gun frame. consider debug"
+	icon_state = "frame_olivaw"
+	generic = FALSE
+	bad_type = /obj/item/part/gun/frame
+	matter = list(MATERIAL_PLASTEEL = 4)
+	rarity_value = 10
+
+	var/result = /obj/item/gun/projectile
+
+	var/grip = /obj/item/part/gun/grip
+	var/grip_attached = FALSE
+
+	var/variant_grip = FALSE
+	var/gripvar1 = /obj/item/part/gun/grip/wood
+	var/gripvar2 = /obj/item/part/gun/grip/black
+	var/gripvar3
+	var/gripvar4
+
+	var/resultvar1 = /obj/item/gun/projectile
+	var/resultvar2 = /obj/item/gun/projectile
+	var/resultvar3
+	var/resultvar4
```

Comment:
This is really not a good way to do this for different results, you can use a list ordered by indexes or an associative list to change the gun depending on the parts used.